core/src/map.cpp

```diff
@@ -282,31 +288,37 @@ void Map::applySceneUpdates(SceneUpdateErrorCallback _onSceneUpdateError) {
         nextScene = impl->nextScene;
     }
 
-    runAsyncTask([nextScene, updates = std::move(updates), _onSceneUpdateError, this](){
+    runAsyncTask([nextScene, updates = std::move(updates), _onSceneReady, this](){
 
-            if (!SceneLoader::applyUpdates(platform, *nextScene, updates, _onSceneUpdateError)) {
+            if (!SceneLoader::applyUpdates(platform, *nextScene, updates)) {
                 LOGW("Scene updates not applied to current scene");
+
+                if (_onSceneReady) {
+                    SceneError err;
+                    if (!nextScene->errors.empty()) { err = nextScene->errors.front(); }
+                    _onSceneReady(false, err);
+                }
                 return;
             }
 
             bool configApplied = SceneLoader::applyConfig(platform, nextScene);
 
-            impl->jobQueue.add([nextScene, configApplied, _onSceneUpdateError, this]() {
+            impl->jobQueue.add([nextScene, configApplied, _onSceneReady, this]() {
                     {
                         std::lock_guard<std::mutex> lock(impl->sceneMutex);
                         if (nextScene == impl->nextScene) {
                             impl->nextScene.reset();
                         } else {
                             // loadScene[Async] was called in the meantime.
+                            if (_onSceneReady) { _onSceneReady(false, {}); }
                             return;
                         }
                     }
-
                     if (configApplied) {
                         auto s = nextScene;
                         impl->setScene(s);
-                        applySceneUpdates(_onSceneUpdateError);
```

Comment:
Similar to loadSceneAsync, possible certain SceneUpdates are queued after a `applySceneUpdate` is called.